Aplib.Core/Intent/Tactics/PrimitiveTactic.cs

```diff
@@ -16,41 +16,55 @@ public class PrimitiveTactic<TBeliefSet> : Tactic<TBeliefSet>
         /// Metadata about this tactic, used to quickly display the tactic in several contexts.
         /// </param>
-        public PrimitiveTactic(IAction<TBeliefSet> action, Metadata? metadata = null)
-            : base(metadata) => _action = action;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="PrimitiveTactic{TBeliefSet}"/> class with the specified action and guard.
-        /// </summary>
         /// <param name="action">The action of the primitive tactic.</param>
         /// <param name="guard">The guard of the primitive tactic.</param>
-        /// <param name="metadata">
-        /// Metadata about this tactic, used to quickly display the tactic in several contexts.
-        /// </param>
-        public PrimitiveTactic(IAction<TBeliefSet> action,
-            System.Func<TBeliefSet, bool> guard,
-            Metadata? metadata = null)
-            : base(guard, metadata) => _action = action;
+        public PrimitiveTactic(Metadata metadata, IAction<TBeliefSet> action, System.Func<TBeliefSet, bool> guard)
+            : base(metadata, guard) => _action = action;
+
+        /// <inheritdoc cref="PrimitiveTactic{TBeliefSet}(Metadata,IAction{TBeliefSet},System.Func{TBeliefSet,bool})"/>
+        public PrimitiveTactic(IAction<TBeliefSet> action, System.Func<TBeliefSet, bool> guard)
+            : this(new Metadata(), action, guard)
+        {
+        }
+
+        /// <inheritdoc cref="PrimitiveTactic{TBeliefSet}(Metadata,IAction{TBeliefSet},System.Func{TBeliefSet,bool})"/>
+        public PrimitiveTactic(Metadata metadata, IAction<TBeliefSet> action) : this(metadata, action, _ => true) { }
+
+        /// <inheritdoc cref="PrimitiveTactic{TBeliefSet}(Metadata,IAction{TBeliefSet},System.Func{TBeliefSet,bool})"/>
+        public PrimitiveTactic(IAction<TBeliefSet> action) : this(new Metadata(), action) { }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="PrimitiveTactic{TBeliefSet}"/> class with the specified action and guard.
+        /// Initializes a new instance of the <see cref="PrimitiveTactic{TBeliefSet}"/> class with the specified action
+        /// and guard.
         /// </summary>
-        /// <param name="queryAction">The queryable action of the primitive tactic.</param>
-        /// <param name="guard">The guard of the primitive tactic.</param>
         /// <param name="metadata">
         /// Metadata about this tactic, used to quickly display the tactic in several contexts.
         /// </param>
-        public PrimitiveTactic(IQueryable<TBeliefSet> queryAction,
-            System.Func<TBeliefSet, bool> guard,
-            Metadata? metadata = null)
-            : base(beliefSet => guard(beliefSet) && queryAction.Query(beliefSet), metadata)
-        => _action = queryAction;
+        /// <param name="queryAction">The queryable action of the primitive tactic.</param>
+        /// <param name="guard">The guard of the primitive tactic.</param>
```

Comment:
You can use inheritdoc and just add the line
`/// <param name="queryAction">The queryable action of the primitive tactic.</param>`

Reply:
I'm not sure if that is a good idea because it takes the Rider bug from being potentially confusing to definitely very confusing. And the parameter ordering is then wrong.

Reply:
You have a point, but I'm not sure if we should work around Rider bugs. The parameter ordering is also a Rider thing. But we can leave it like this.

Reply:
Could you test the parameter ordering in docfx?